GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.DownloadObject.cs

```diff
@@ -0,0 +1,165 @@
+        // See issue #21.
+
+        public void DownloadObject(string bucket, string sourceName, Stream destination)
+        {
+            DownloadObject(bucket, sourceName, destination, options: null, progress: null);
+        }
+
+        public void DownloadObject(
+            string bucket,
+            string sourceName,
+            Stream destination,
+            DownloadObjectOptions options,
+            IProgress<IDownloadProgress> progress)
+        {
+            string uri = GetUri(bucket, sourceName);
+            DownloadObjectImpl(uri, destination, options, progress);
+        }
+
+        public Task DownloadObjectAsync(string bucket, string sourceName, Stream destination)
+        {
+            return DownloadObjectAsync(bucket, sourceName, destination,
+                options: null, cancellationToken: CancellationToken.None, progress: null);
+        }
+
+        public Task DownloadObjectAsync(
+            string bucket,
+            string sourceName,
+            Stream destination,
+            DownloadObjectOptions options,
+            CancellationToken cancellationToken,
+            IProgress<IDownloadProgress> progress)
+        {
+            string uri = GetUri(bucket, sourceName);
+            return DownloadObjectAsyncImpl(uri, destination, options, cancellationToken, progress);
+        }
+
+        public Task DownloadObjectAsync(Object source, Stream destination)
+        {
+            return DownloadObjectAsync(source, destination,
+                options: null, cancellationToken: CancellationToken.None, progress: null);
+        }
+
+        public void DownloadObject(Object source, Stream destination)
+        {
+            DownloadObject(source, destination, options: null, progress: null);
+        }
+
+        public void DownloadObject(
+            Object source,
+            Stream destination,
+            DownloadObjectOptions options,
+            IProgress<IDownloadProgress> progress)
+        {
+            string uri = GetUri(source);
+            DownloadObjectImpl(uri, destination, options, progress);
+        }
+
+        public Task DownloadObjectAsync(
+            Object source,
+            Stream destination,
+            DownloadObjectOptions options,
+            CancellationToken cancellationToken,
+            IProgress<IDownloadProgress> progress)
+        {
+            string uri = GetUri(source);
+            return DownloadObjectAsyncImpl(uri, destination, options, cancellationToken, progress);
+        }
+
+        /// <summary>
+        /// Constructs the media URL of an object from its bucket and name.
+        /// </summary>
+        private static string GetUri(string bucket, string sourceName)
+        {
+            bucket.CheckNotNull(nameof(bucket));
+            sourceName.CheckNotNull(nameof(sourceName));
+            return $"https://www.googleapis.com/download/storage/v1/b/{Uri.EscapeDataString(bucket)}/o/{Uri.EscapeDataString(sourceName)}?alt=media";
```

Comment:
I don't think that the bucket needs to be escaped, only dns-like names are valid bucket names. Although having a precondition that the bucket name is indeed valid would not hurt.